src/graphics/vulkan/vulkan_texture.cpp

```
@@ -78,48 +83,43 @@ namespace moth_graphics::graphics::vulkan {
+        UniqueHandle<VkSampler>& slot = nearest ? m_vkSamplerNearest : m_vkSamplerLinear;
+        if (!slot) {
+            VkDevice const device = m_context.GetVkDevice();
+            slot = UniqueHandle<VkSampler>(CreateSampler(m_minFilter, m_magFilter), [device](VkSampler h) {
+                vkDestroySampler(device, h, nullptr);
+            });
         }
         return slot;
     }
 
     VkDescriptorSet Texture::GetDescriptorSet() const {
         VkSampler const currentSampler = GetVkSampler();
```

Clang (14.0.6)

[error] variable 'currentSampler' is not initialized

(cppcoreguidelines-init-variables,-warnings-as-errors)